tests/infra_test.go

```diff
@@ -71,6 +74,104 @@ var _ = Describe("Infrastructure", func() {
 		return tests.WaitForSuccessfulVMIStart(obj)
 	}
 
+	Describe("Taints and toleration", func() {
+
+		It("should tolerate CriticalAddonsOnly toleration", func() {
+
+			var kvPods *k8sv1.PodList
+			By("finding all nodes that are running kubevirt components", func() {
+				kvPods, err = virtClient.CoreV1().
+					Pods(tests.KubeVirtInstallNamespace).List(metav1.ListOptions{})
+				Expect(err).ShouldNot(HaveOccurred(), "failed listing kubevirt pods")
+				Expect(len(kvPods.Items)).
+					To(BeNumerically(">", 0), "no kubevirt pods found")
+			})
+
+			// nodes that run kubevirt components.
+			kvNodes := make(map[string]*k8sv1.Node)
+			By("adding taints to any node that runs kubevirt component", func() {
+				criticalPodTaint := k8sv1.Taint{
+					Key:    "CriticalAddonsOnly",
+					Value:  "",
+					Effect: k8sv1.TaintEffectNoExecute,
+				}
+
+				for _, kvPod := range kvPods.Items {
+					kvNode, err := virtClient.CoreV1().Nodes().Get(kvPod.Spec.NodeName, metav1.GetOptions{})
```

Comment:
```suggestion
                                      if kvNodes[kvPod.Spec.NodeName] != nil { // small optimization
                                          continue
                                      }
					kvNode, err := virtClient.CoreV1().Nodes().Get(kvPod.Spec.NodeName, metav1.GetOptions{})
```

Reply:
@dhiller hey, I'm not sure I understand the suggestion :)